public/video-ui/src/slices/usage.ts

```diff
@@ -0,0 +1,107 @@
+import { Action, createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import ErrorMessages from '../constants/ErrorMessages';
+import { CapiContent, Stage as CapiStage } from '../services/capi';
+import VideosApi from '../services/VideosApi';
+import { showError } from './error';
+
+
+export type UsageData = Record<CapiStage, {
+    video: CapiContent[];
+    other: CapiContent[];
+}>;
+
+export type UsageState = {
+    data: UsageData;
+    totalUsages: number,
+    totalVideoPages: number,
+    isFetching: boolean,
+}
+
+const getInitialState = (): UsageState => ({
+    data: {
+        published: {
+            video: [],
+            other: []
+        },
+        preview: {
+            video: [],
+            other: []
+        }
+    },
+
+    totalUsages: 0,
+    totalVideoPages: 0,
+    isFetching: false
+});
+
+export const fetchUsages = createAsyncThunk<
+    UsageState,
+    string
+>(
+    'usage/fetchUsages',
+    (id, { dispatch }) => {
+        return VideosApi.getVideoUsages(id).then(
+            data => {
+                const { preview, published } = data;
+                // a lot of components conditionally render based on the number of usages,
+                // rather than constantly call a utility function, they use the stateful value.
+                // It should be possible to refactor those components to derive the number
+                // from the data so the value does not need to be in the UsageState.
```

Comment:
This sounds like a nice improvement. In the short term can we change the payload of the action to only include the data and have the logic to determine the totals in the reducer? 

https://redux.js.org/style-guide/#put-as-much-logic-as-possible-in-reducers

Ultimately it would be nice to have the total provided as a selector.

Reply:
👍  I had shied away from doing that as it would involve changing the implementation of saveVideo.js (since it uses the action returned by `fetchUsages`), but after converting it to typescript, that seemed less risky